### PR TITLE
Fix GCC 7 (use snprintf, inc. buffer size, remove not needed code)

### PR DESCRIPTION
**Pull request by davidlt**
Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

### CODE REVIEW

**Comment by davidlt** [comment]
please test

File: RecoMuon/StandAloneTrackFinder/src/StandAloneTrajectoryBuilder.cc [hunk 1]
```diff
@@ -278,11 +278,7 @@ StandAloneMuonTrajectoryBuilder::trajectories(const TrajectorySeed& seed){
   
   TrajectorySeed seedForBW;
 
-  if(theBWSeedType == "noSeed") {
-    TrajectorySeed seedVoid;
-    seedForBW = seedVoid;
-  }
-  else if(theBWSeedType == "fromFWFit") {
+  if(theBWSeedType == "fromFWFit") {
```
Comment (slava77) [inline RecoMuon/StandAloneTrackFinder/src/StandAloneTrajectoryBuilder.cc]: this looks incorrect: "noSeed" will now go to the final "else" with a LogWarning.
The expected behavior should the that ``` theBWSeedType == "noSeed" ``` is a no-op

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17457	f38740f
- technical changes; includes a replacement to #17276
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by davidlange6** [comment]
+1

**Pull request merged**